source/Core/BSP/Pinecilv2/IRQ.cpp

```diff
@@ -26,19 +26,18 @@ void                                 adc_fifo_irq(void) {
   if (ADC_GetIntStatus(ADC_INT_FIFO_READY) == SET) {
     // Read out all entries in the fifo
     while (ADC_Get_FIFO_Count()) {
-      volatile uint32_t reading = ADC_Read_FIFO();
-      // As per manual, 26 bit reading; lowest 16 are the ADC
-      uint16_t sample = reading & 0xFFFF;
-      uint8_t  source = (reading >> 21) & 0b11111;
-      switch (source) {
+      uint32_t reading = ADC_Read_FIFO();
+      ADC_Result_Type parsed;
+      ADC_Parse_Result(&reading, 1, &parsed);
+      switch (parsed.posChan) {
       case TMP36_ADC_CHANNEL:
-        ADC_Temp.update(sample);
+        ADC_Temp.update(parsed.value * 4);

```

Comment:
If the intend of the `*4` is to shift to the left by 2 bits; it would be clearer to use the shift operator?

Reply:
done